File: app/src/main/java/net/mbonnin/arcanetracker/parser/Player.java
package net.mbonnin.arcanetracker.parser;

import net.mbonnin.arcanetracker.CardUtil;
import net.mbonnin.arcanetracker.HeroUtilKt;
import net.mbonnin.hsmodel.Card;

/**
 * Created by martin on 11/7/16.
 */
public class Player {
    public Entity entity;
    public String battleTag;
    public boolean isOpponent;
    public boolean hasCoin;

    public Entity hero;
    public Entity heroPower;

    public int classIndex() {
        Card card = CardUtil.INSTANCE.getCard(hero.CardID);
        return HeroUtilKt.getClassIndex(card.playerClass);
    }


    public String playerClass() {
        Card card = CardUtil.INSTANCE.getCard(hero.CardID);
        return card.playerClass;
    }
}

File: app/src/main/java/net/mbonnin/arcanetracker/parser/Player.kt
package net.mbonnin.arcanetracker.parser

import net.mbonnin.arcanetracker.CardUtil
import net.mbonnin.arcanetracker.getClassIndex

/**
 * Created by martin on 11/7/16.
 */
class Player {
    var entity: Entity? = null
    var battleTag: String? = null
    var isOpponent: Boolean = false
    var hasCoin: Boolean = false

    var hero: Entity? = null
    var heroPower: Entity? = null

    fun classIndex(): Int {
        val (_, _, _, playerClass) = CardUtil.getCard(hero!!.CardID!!)
        return getClassIndex(playerClass)
    }


    fun playerClass(): String {
        val (_, _, _, playerClass) = CardUtil.getCard(hero!!.CardID!!)
        return playerClass
    }
}
